pkg/loadbalancer/writer/writer.go

```diff
@@ -264,14 +269,53 @@ func (w *Writer) UpdateBackendHealth(txn WriteTxn, serviceName loadbalancer.Serv
 	return true, w.RefreshFrontends(txn, serviceName)
 }
 
+func (w *Writer) isServiceLBIPAM(svc *loadbalancer.Service) bool {
+	svcLBClass := ""
+	if svc.LoadBalancerClass != nil {
+		svcLBClass = *svc.LoadBalancerClass
+	}
+
+	switch w.externalConfig.DefaultLBServiceIPAM {
+	case option.ServiceLBClassLBIPAM:
+		// The default allocator is LB-IPAM, so we check the Service LB Class
+		// is not overriding LB-IPAM.
+		return (svcLBClass != constants.ServiceLBClassNodeIPAM)
+
+	case option.ServiceLBClassNodeIPAM:
+		// TODO: at the time of writing there's no obvious way to configure a
+		// Service VIP to be allocated via LB-IPAM, if the default is Node-IPAM.
+	}
+
+	return false
```

Comment:
Something like this should cover everything I think:
(I wrote this in github comment directly so there might be some typo here and there)

```suggestion
    if svc.LoadBalancerClass == nil {
        return w.externalConfig.DefaultLBServiceIPAM == option.ServiceLBClassLBIPAM
    }
    return *svc.LoadBalancerClass == cilium_api_v2alpha1.BGPLoadBalancerClass || *svc.LoadBalancerClass == cilium_api_v2alpha1.L2AnnounceLoadBalancerClass 
```

This also make it only LB IPAM specific which might lift the requirement to pull the node ipam annotation outside of its package I think?